RuntimeError: Illegal character '&#55357;' in raw string "java.lang.IllegalStateException...

We are attempting to use this script to export email items into PST. Unfortunately we are coming across illegal characters in some email subjects that are preventing the script from running successfully. Any suggestions or fixes possible? I think one option would be moving these items to an "Error items" folder and continuing with the remaining items into PST or substituting these issue characters with an `_` (or some other character).

**Full error output:**

```
2021-05-07 09:47:47: Getting items to export
2021-05-07 09:47:47: 90219 selected items
2021-05-07 09:47:49: 90219 top-level items to export
2021-05-07 09:47:57: Exporting to T:\XXXCUSTODIANXXX
2021-05-07 09:48:00: Exporting custodian: XXXCUSTODIANXXX (90219 items)
2021-05-07 10:30:24: Renaming files
2021-05-07 10:30:24: Creating T:\XXXCUSTODIANXXX/Reports/XXXCUSTODIANXXX/summary-report.txt
2021-05-07 10:30:24: Creating T:\XXXCUSTODIANXXX/Reports/XXXCUSTODIANXXX/summary-report.xml
2021-05-07 10:30:24: Creating T:\XXXCUSTODIANXXX/Reports/XXXCUSTODIANXXX/top-level-MD5-digests.txt
2021-05-07 10:30:24: Exported 90219 of 90219 items
2021-05-07 10:30:24: Summarizing reports in T:\XXXCUSTODIANXXX/Reports
2021-05-07 10:30:24: Reading T:\XXXCUSTODIANXXX/Reports/XXXCUSTODIANXXX/summary-report.xml
2021-05-07 10:30:38: Error occured in provided block:
2021-05-07 10:30:39: (ParseException) #<RuntimeError: Illegal character '&#55357;' in raw string "java.lang.IllegalStateException: Error exporting item into mailbox: ExportItemContext[guid=8342217eede74b3ea3cae285b397ff44,mime-type=application/vnd.ms-outlook-note,item=ItemDetails{name=&#55357;&#56798; Twitch bits x 3, container=false, excluded=false, failure=false, flags=[TEXT_INDEXED, AUDITED, TOP_LEVEL, NOT_LOOSE_FILE, NOT_PHYSICAL_FILE, FAMILY_INLINE, NOT_FILE_DATA, NOT_FAILED, E_DISCOVERY_CLASS]},numberingContext=NumberingExportContext[numbering=001.047.981,documentNum=001.047.981],nativeContext=NativeExportContext[fromBinaryStore=false,isSlipSheet=false,retainAsSlipSheet=false,native=EXPORT_DIRECTORY/XXXCUSTODIANXXX/XXXCUSTODIANXXX [O365_Emails_20210111-20210503]/XXXCUSTODIANXXX [O365_Emails_20210111-20210503]/XXXCUSTODIANXXX.pst/Top of Personal Folders/XXXCUSTODIANXXX (Primary)/Recoverable Items/Deletions/&#55357;&#56798; Twitch bits x 3.msg,original=EXPORT_SPOOL/834/8342217eede74b3ea3cae285b397ff44-ORIGINAL.msg,tmpNative=EXPORT_SPOOL/834/8342217eede74b3ea3cae285b397ff44-ORIGINAL.msg,replaceWithSlipsheet=false],pdfContext=PdfExportContext[pdfPageCount=0,stampedPageCount=0,storedInPrintStore=false,pdf=&lt;null&gt;,tmpPdf=&lt;null&gt;,imagingProfileName=&lt;null&gt;],productionProfileContext=ProductionProfileExportContext[profileName=&lt;null&gt;,ruleName=&lt;null&gt;],tiffContext=ImageExportContext[imagePageCount=0,super=ImageExportContext[filesProduced=]],textContext=TextExportContext[slipSheetText=false,hasCommunicationAspect=false]]
	at com.nuix.export.processor.J.a(SourceFile:338)
	at com.nuix.export.processor.J.a(SourceFile:237)
	at com.nuix.processing.worker.handler.export.i.a(SourceFile:109)
	at com.nuix.processing.worker.handler.export.i.handleMessage(SourceFile:26)
	at com.nuix.processing.worker.Worker.processMessage(SourceFile:547)
	at com.nuix.processing.worker.Worker.execute(SourceFile:466)
	at com.nuix.processing.workerlauncher.SingleWorkerMain.a(SourceFile:305)
	at com.nuix.processing.workerlauncher.SingleWorkerMain.g(SourceFile:143)
	at com.nuix.bootstrap.c$b.a(SourceFile:259)
	at com.nuix.bootstrap.c.run(SourceFile:118)
	at com.nuix.processing.workerlauncher.SingleWorkerMain.main(SourceFile:383)
Caused by: AsposeException: Header CRC is not valid !
[Ljava.lang.StackTraceElement;@651e0218
	at com.aspose.email.zang.j(SourceFile:2033)
	at com.aspose.email.zang.a(SourceFile:1554)
	at com.aspose.email.zang.&lt;init&gt;(SourceFile:170)
	at com.aspose.email.zafo.&lt;init&gt;(SourceFile:71)
	at com.aspose.email.zaso.&lt;init&gt;(SourceFile:132)
	at com.aspose.email.PersonalStorage.a(SourceFile:277)
	at com.aspose.email.PersonalStorage.fromFile(SourceFile:237)
	at com.aspose.email.PersonalStorage.fromFile(SourceFile:211)
	at com.nuix.export.mail.mailbox.a.&lt;init&gt;(SourceFile:63)
	at com.nuix.export.mail.mailbox.d.&lt;init&gt;(SourceFile:26)
	at com.nuix.export.mail.mailbox.t.a(SourceFile:163)
	at com.nuix.export.mail.mailbox.u.a(SourceFile:171)
	at com.nuix.export.mail.mailbox.e.exportMessage(SourceFile:85)
	at com.nuix.export.processor.J.a(SourceFile:325)
	... 10 more
">
uri:classloader:/META-INF/jruby.home/lib/ruby/stdlib/rexml/text.rb:162:in `block in check'
org/jruby/RubyString.java:4350:in `scan'
uri:classloader:/META-INF/jruby.home/lib/ruby/stdlib/rexml/text.rb:154:in `check'
uri:classloader:/META-INF/jruby.home/lib/ruby/stdlib/rexml/attribute.rb:157:in `element='
uri:classloader:/META-INF/jruby.home/lib/ruby/stdlib/rexml/element.rb:1125:in `[]='
uri:classloader:/META-INF/jruby.home/lib/ruby/stdlib/rexml/parsers/treeparser.rb:36:in `block in parse'
org/jruby/RubyHash.java:1419:in `each'
uri:classloader:/META-INF/jruby.home/lib/ruby/stdlib/rexml/parsers/treeparser.rb:35:in `parse'
uri:classloader:/META-INF/jruby.home/lib/ruby/stdlib/rexml/document.rb:288:in `build'
uri:classloader:/META-INF/jruby.home/lib/ruby/stdlib/rexml/document.rb:45:in `initialize'
C:/Users/Administrator.GIFS/AppData/Roaming/Nuix/Scripts/NuixProvided/CustodianPstExport.nuixscript/summary_reporter.rb:192:in `initialize'
C:/Users/Administrator.GIFS/AppData/Roaming/Nuix/Scripts/NuixProvided/CustodianPstExport.nuixscript/summary_reporter.rb:87:in `summarize_file'
C:/Users/Administrator.GIFS/AppData/Roaming/Nuix/Scripts/NuixProvided/CustodianPstExport.nuixscript/summary_reporter.rb:76:in `block in summarize'
org/jruby/RubyArray.java:1792:in `each'
org/jruby/RubyEnumerable.java:1194:in `each_with_index'
C:/Users/Administrator.GIFS/AppData/Roaming/Nuix/Scripts/NuixProvided/CustodianPstExport.nuixscript/summary_reporter.rb:75:in `summarize'
C:/Users/Administrator.GIFS/AppData/Roaming/Nuix/Scripts/NuixProvided/CustodianPstExport.nuixscript/summary_reporter.rb:21:in `initialize'
C:\Users\Administrator.GIFS\AppData\Roaming\Nuix\Scripts\NuixProvided\CustodianPstExport.nuixscript\custodian_pst.rb:202:in `run'
C:\Users\Administrator.GIFS\AppData\Roaming\Nuix\Scripts\NuixProvided\CustodianPstExport.nuixscript\custodian_pst.rb:38:in `block in initialize'
C:\Users\Administrator.GIFS\AppData\Roaming\Nuix\Scripts\NuixProvided\CustodianPstExport.nuixscript\custodian_pst.rb:34:in `initialize'
C:\Users\Administrator.GIFS\AppData\Roaming\Nuix\Scripts\NuixProvided\CustodianPstExport.nuixscript\custodian_pst.rb:235:in `<main>'
...
Illegal character '&#55357;' in raw string "java.lang.IllegalStateException: Error exporting item into mailbox: ExportItemContext[guid=8342217eede74b3ea3cae285b397ff44,mime-type=application/vnd.ms-outlook-note,item=ItemDetails{name=&#55357;&#56798; Twitch bits x 3, container=false, excluded=false, failure=false, flags=[TEXT_INDEXED, AUDITED, TOP_LEVEL, NOT_LOOSE_FILE, NOT_PHYSICAL_FILE, FAMILY_INLINE, NOT_FILE_DATA, NOT_FAILED, E_DISCOVERY_CLASS]},numberingContext=NumberingExportContext[numbering=001.047.981,documentNum=001.047.981],nativeContext=NativeExportContext[fromBinaryStore=false,isSlipSheet=false,retainAsSlipSheet=false,native=EXPORT_DIRECTORY/XXXCUSTODIANXXX/XXXCUSTODIANXXX [O365_Emails_20210111-20210503]/XXXCUSTODIANXXX [O365_Emails_20210111-20210503]/XXXCUSTODIANXXX.pst/Top of Personal Folders/XXXCUSTODIANXXX (Primary)/Recoverable Items/Deletions/&#55357;&#56798; Twitch bits x 3.msg,original=EXPORT_SPOOL/834/8342217eede74b3ea3cae285b397ff44-ORIGINAL.msg,tmpNative=EXPORT_SPOOL/834/8342217eede74b3ea3cae285b397ff44-ORIGINAL.msg,replaceWithSlipsheet=false],pdfContext=PdfExportContext[pdfPageCount=0,stampedPageCount=0,storedInPrintStore=false,pdf=&lt;null&gt;,tmpPdf=&lt;null&gt;,imagingProfileName=&lt;null&gt;],productionProfileContext=ProductionProfileExportContext[profileName=&lt;null&gt;,ruleName=&lt;null&gt;],tiffContext=ImageExportContext[imagePageCount=0,super=ImageExportContext[filesProduced=]],textContext=TextExportContext[slipSheetText=false,hasCommunicationAspect=false]]
	at com.nuix.export.processor.J.a(SourceFile:338)
	at com.nuix.export.processor.J.a(SourceFile:237)
	at com.nuix.processing.worker.handler.export.i.a(SourceFile:109)
	at com.nuix.processing.worker.handler.export.i.handleMessage(SourceFile:26)
	at com.nuix.processing.worker.Worker.processMessage(SourceFile:547)
	at com.nuix.processing.worker.Worker.execute(SourceFile:466)
	at com.nuix.processing.workerlauncher.SingleWorkerMain.a(SourceFile:305)
	at com.nuix.processing.workerlauncher.SingleWorkerMain.g(SourceFile:143)
	at com.nuix.bootstrap.c$b.a(SourceFile:259)
	at com.nuix.bootstrap.c.run(SourceFile:118)
	at com.nuix.processing.workerlauncher.SingleWorkerMain.main(SourceFile:383)
Caused by: AsposeException: Header CRC is not valid !
[Ljava.lang.StackTraceElement;@651e0218
	at com.aspose.email.zang.j(SourceFile:2033)
	at com.aspose.email.zang.a(SourceFile:1554)
	at com.aspose.email.zang.&lt;init&gt;(SourceFile:170)
	at com.aspose.email.zafo.&lt;init&gt;(SourceFile:71)
	at com.aspose.email.zaso.&lt;init&gt;(SourceFile:132)
	at com.aspose.email.PersonalStorage.a(SourceFile:277)
	at com.aspose.email.PersonalStorage.fromFile(SourceFile:237)
	at com.aspose.email.PersonalStorage.fromFile(SourceFile:211)
	at com.nuix.export.mail.mailbox.a.&lt;init&gt;(SourceFile:63)
	at com.nuix.export.mail.mailbox.d.&lt;init&gt;(SourceFile:26)
	at com.nuix.export.mail.mailbox.t.a(SourceFile:163)
	at com.nuix.export.mail.mailbox.u.a(SourceFile:171)
	at com.nuix.export.mail.mailbox.e.exportMessage(SourceFile:85)
	at com.nuix.export.processor.J.a(SourceFile:325)
	... 10 more
"
Line: 518896
Position: 49857113
Last 80 unconsumed characters:

2021-05-07 10:30:39: org.jruby.exceptions.RuntimeError: (ParseException) #<RuntimeError: Illegal character '&#55357;' in raw string "java.lang.IllegalStateException: Error exporting item into mailbox: ExportItemContext[guid=8342217eede74b3ea3cae285b397ff44,mime-type=application/vnd.ms-outlook-note,item=ItemDetails{name=&#55357;&#56798; Twitch bits x 3, container=false, excluded=false, failure=false, flags=[TEXT_INDEXED, AUDITED, TOP_LEVEL, NOT_LOOSE_FILE, NOT_PHYSICAL_FILE, FAMILY_INLINE, NOT_FILE_DATA, NOT_FAILED, E_DISCOVERY_CLASS]},numberingContext=NumberingExportContext[numbering=001.047.981,documentNum=001.047.981],nativeContext=NativeExportContext[fromBinaryStore=false,isSlipSheet=false,retainAsSlipSheet=false,native=EXPORT_DIRECTORY/XXXCUSTODIANXXX/XXXCUSTODIANXXX [O365_Emails_20210111-20210503]/XXXCUSTODIANXXX [O365_Emails_20210111-20210503]/XXXCUSTODIANXXX.pst/Top of Personal Folders/XXXCUSTODIANXXX (Primary)/Recoverable Items/Deletions/&#55357;&#56798; Twitch bits x 3.msg,original=EXPORT_SPOOL/834/8342217eede74b3ea3cae285b397ff44-ORIGINAL.msg,tmpNative=EXPORT_SPOOL/834/8342217eede74b3ea3cae285b397ff44-ORIGINAL.msg,replaceWithSlipsheet=false],pdfContext=PdfExportContext[pdfPageCount=0,stampedPageCount=0,storedInPrintStore=false,pdf=&lt;null&gt;,tmpPdf=&lt;null&gt;,imagingProfileName=&lt;null&gt;],productionProfileContext=ProductionProfileExportContext[profileName=&lt;null&gt;,ruleName=&lt;null&gt;],tiffContext=ImageExportContext[imagePageCount=0,super=ImageExportContext[filesProduced=]],textContext=TextExportContext[slipSheetText=false,hasCommunicationAspect=false]]
	at com.nuix.export.processor.J.a(SourceFile:338)
	at com.nuix.export.processor.J.a(SourceFile:237)
	at com.nuix.processing.worker.handler.export.i.a(SourceFile:109)
	at com.nuix.processing.worker.handler.export.i.handleMessage(SourceFile:26)
	at com.nuix.processing.worker.Worker.processMessage(SourceFile:547)
	at com.nuix.processing.worker.Worker.execute(SourceFile:466)
	at com.nuix.processing.workerlauncher.SingleWorkerMain.a(SourceFile:305)
	at com.nuix.processing.workerlauncher.SingleWorkerMain.g(SourceFile:143)
	at com.nuix.bootstrap.c$b.a(SourceFile:259)
	at com.nuix.bootstrap.c.run(SourceFile:118)
	at com.nuix.processing.workerlauncher.SingleWorkerMain.main(SourceFile:383)
Caused by: AsposeException: Header CRC is not valid !
[Ljava.lang.StackTraceElement;@651e0218
	at com.aspose.email.zang.j(SourceFile:2033)
	at com.aspose.email.zang.a(SourceFile:1554)
	at com.aspose.email.zang.&lt;init&gt;(SourceFile:170)
	at com.aspose.email.zafo.&lt;init&gt;(SourceFile:71)
	at com.aspose.email.zaso.&lt;init&gt;(SourceFile:132)
	at com.aspose.email.PersonalStorage.a(SourceFile:277)
	at com.aspose.email.PersonalStorage.fromFile(SourceFile:237)
	at com.aspose.email.PersonalStorage.fromFile(SourceFile:211)
	at com.nuix.export.mail.mailbox.a.&lt;init&gt;(SourceFile:63)
	at com.nuix.export.mail.mailbox.d.&lt;init&gt;(SourceFile:26)
	at com.nuix.export.mail.mailbox.t.a(SourceFile:163)
	at com.nuix.export.mail.mailbox.u.a(SourceFile:171)
	at com.nuix.export.mail.mailbox.e.exportMessage(SourceFile:85)
	at com.nuix.export.processor.J.a(SourceFile:325)
	... 10 more
">
uri:classloader:/META-INF/jruby.home/lib/ruby/stdlib/rexml/text.rb:162:in `block in check'
org/jruby/RubyString.java:4350:in `scan'
uri:classloader:/META-INF/jruby.home/lib/ruby/stdlib/rexml/text.rb:154:in `check'
uri:classloader:/META-INF/jruby.home/lib/ruby/stdlib/rexml/attribute.rb:157:in `element='
uri:classloader:/META-INF/jruby.home/lib/ruby/stdlib/rexml/element.rb:1125:in `[]='
uri:classloader:/META-INF/jruby.home/lib/ruby/stdlib/rexml/parsers/treeparser.rb:36:in `block in parse'
org/jruby/RubyHash.java:1419:in `each'
uri:classloader:/META-INF/jruby.home/lib/ruby/stdlib/rexml/parsers/treeparser.rb:35:in `parse'
uri:classloader:/META-INF/jruby.home/lib/ruby/stdlib/rexml/document.rb:288:in `build'
uri:classloader:/META-INF/jruby.home/lib/ruby/stdlib/rexml/document.rb:45:in `initialize'
C:/Users/Administrator.GIFS/AppData/Roaming/Nuix/Scripts/NuixProvided/CustodianPstExport.nuixscript/summary_reporter.rb:192:in `initialize'
C:/Users/Administrator.GIFS/AppData/Roaming/Nuix/Scripts/NuixProvided/CustodianPstExport.nuixscript/summary_reporter.rb:87:in `summarize_file'
C:/Users/Administrator.GIFS/AppData/Roaming/Nuix/Scripts/NuixProvided/CustodianPstExport.nuixscript/summary_reporter.rb:76:in `block in summarize'
org/jruby/RubyArray.java:1792:in `each'
org/jruby/RubyEnumerable.java:1194:in `each_with_index'
C:/Users/Administrator.GIFS/AppData/Roaming/Nuix/Scripts/NuixProvided/CustodianPstExport.nuixscript/summary_reporter.rb:75:in `summarize'
C:/Users/Administrator.GIFS/AppData/Roaming/Nuix/Scripts/NuixProvided/CustodianPstExport.nuixscript/summary_reporter.rb:21:in `initialize'
C:\Users\Administrator.GIFS\AppData\Roaming\Nuix\Scripts\NuixProvided\CustodianPstExport.nuixscript\custodian_pst.rb:202:in `run'
C:\Users\Administrator.GIFS\AppData\Roaming\Nuix\Scripts\NuixProvided\CustodianPstExport.nuixscript\custodian_pst.rb:38:in `block in initialize'
C:\Users\Administrator.GIFS\AppData\Roaming\Nuix\Scripts\NuixProvided\CustodianPstExport.nuixscript\custodian_pst.rb:34:in `initialize'
C:\Users\Administrator.GIFS\AppData\Roaming\Nuix\Scripts\NuixProvided\CustodianPstExport.nuixscript\custodian_pst.rb:235:in `<main>'
...
Illegal character '&#55357;' in raw string "java.lang.IllegalStateException: Error exporting item into mailbox: ExportItemContext[guid=8342217eede74b3ea3cae285b397ff44,mime-type=application/vnd.ms-outlook-note,item=ItemDetails{name=&#55357;&#56798; Twitch bits x 3, container=false, excluded=false, failure=false, flags=[TEXT_INDEXED, AUDITED, TOP_LEVEL, NOT_LOOSE_FILE, NOT_PHYSICAL_FILE, FAMILY_INLINE, NOT_FILE_DATA, NOT_FAILED, E_DISCOVERY_CLASS]},numberingContext=NumberingExportContext[numbering=001.047.981,documentNum=001.047.981],nativeContext=NativeExportContext[fromBinaryStore=false,isSlipSheet=false,retainAsSlipSheet=false,native=EXPORT_DIRECTORY/XXXCUSTODIANXXX/XXXCUSTODIANXXX [O365_Emails_20210111-20210503]/XXXCUSTODIANXXX [O365_Emails_20210111-20210503]/XXXCUSTODIANXXX.pst/Top of Personal Folders/XXXCUSTODIANXXX (Primary)/Recoverable Items/Deletions/&#55357;&#56798; Twitch bits x 3.msg,original=EXPORT_SPOOL/834/8342217eede74b3ea3cae285b397ff44-ORIGINAL.msg,tmpNative=EXPORT_SPOOL/834/8342217eede74b3ea3cae285b397ff44-ORIGINAL.msg,replaceWithSlipsheet=false],pdfContext=PdfExportContext[pdfPageCount=0,stampedPageCount=0,storedInPrintStore=false,pdf=&lt;null&gt;,tmpPdf=&lt;null&gt;,imagingProfileName=&lt;null&gt;],productionProfileContext=ProductionProfileExportContext[profileName=&lt;null&gt;,ruleName=&lt;null&gt;],tiffContext=ImageExportContext[imagePageCount=0,super=ImageExportContext[filesProduced=]],textContext=TextExportContext[slipSheetText=false,hasCommunicationAspect=false]]
	at com.nuix.export.processor.J.a(SourceFile:338)
	at com.nuix.export.processor.J.a(SourceFile:237)
	at com.nuix.processing.worker.handler.export.i.a(SourceFile:109)
	at com.nuix.processing.worker.handler.export.i.handleMessage(SourceFile:26)
	at com.nuix.processing.worker.Worker.processMessage(SourceFile:547)
	at com.nuix.processing.worker.Worker.execute(SourceFile:466)
	at com.nuix.processing.workerlauncher.SingleWorkerMain.a(SourceFile:305)
	at com.nuix.processing.workerlauncher.SingleWorkerMain.g(SourceFile:143)
	at com.nuix.bootstrap.c$b.a(SourceFile:259)
	at com.nuix.bootstrap.c.run(SourceFile:118)
	at com.nuix.processing.workerlauncher.SingleWorkerMain.main(SourceFile:383)
Caused by: AsposeException: Header CRC is not valid !
[Ljava.lang.StackTraceElement;@651e0218
	at com.aspose.email.zang.j(SourceFile:2033)
	at com.aspose.email.zang.a(SourceFile:1554)
	at com.aspose.email.zang.&lt;init&gt;(SourceFile:170)
	at com.aspose.email.zafo.&lt;init&gt;(SourceFile:71)
	at com.aspose.email.zaso.&lt;init&gt;(SourceFile:132)
	at com.aspose.email.PersonalStorage.a(SourceFile:277)
	at com.aspose.email.PersonalStorage.fromFile(SourceFile:237)
	at com.aspose.email.PersonalStorage.fromFile(SourceFile:211)
	at com.nuix.export.mail.mailbox.a.&lt;init&gt;(SourceFile:63)
	at com.nuix.export.mail.mailbox.d.&lt;init&gt;(SourceFile:26)
	at com.nuix.export.mail.mailbox.t.a(SourceFile:163)
	at com.nuix.export.mail.mailbox.u.a(SourceFile:171)
	at com.nuix.export.mail.mailbox.e.exportMessage(SourceFile:85)
	at com.nuix.export.processor.J.a(SourceFile:325)
	... 10 more
"
Line: 518896
Position: 49857113
Last 80 unconsumed characters:

	at RUBY.parse(uri:classloader:/META-INF/jruby.home/lib/ruby/stdlib/rexml/parsers/treeparser.rb:96)
	at RUBY.build(uri:classloader:/META-INF/jruby.home/lib/ruby/stdlib/rexml/document.rb:288)
	at RUBY.initialize(uri:classloader:/META-INF/jruby.home/lib/ruby/stdlib/rexml/document.rb:45)
	at RUBY.initialize(C:/Users/Administrator.GIFS/AppData/Roaming/Nuix/Scripts/NuixProvided/CustodianPstExport.nuixscript/summary_reporter.rb:192)
	at RUBY.summarize_file(C:/Users/Administrator.GIFS/AppData/Roaming/Nuix/Scripts/NuixProvided/CustodianPstExport.nuixscript/summary_reporter.rb:87)
	at RUBY.summarize(C:/Users/Administrator.GIFS/AppData/Roaming/Nuix/Scripts/NuixProvided/CustodianPstExport.nuixscript/summary_reporter.rb:76)
	at org.jruby.RubyArray.each(org/jruby/RubyArray.java:1792)
	at org.jruby.RubyEnumerable.each_with_index(org/jruby/RubyEnumerable.java:1194)
	at RUBY.summarize(C:/Users/Administrator.GIFS/AppData/Roaming/Nuix/Scripts/NuixProvided/CustodianPstExport.nuixscript/summary_reporter.rb:75)
	at RUBY.initialize(C:/Users/Administrator.GIFS/AppData/Roaming/Nuix/Scripts/NuixProvided/CustodianPstExport.nuixscript/summary_reporter.rb:21)
	at RUBY.run(C:\Users\Administrator.GIFS\AppData\Roaming\Nuix\Scripts\NuixProvided\CustodianPstExport.nuixscript\custodian_pst.rb:202)
	at RUBY.initialize(C:\Users\Administrator.GIFS\AppData\Roaming\Nuix\Scripts\NuixProvided\CustodianPstExport.nuixscript\custodian_pst.rb:38)
	at RUBY.initialize(C:\Users\Administrator.GIFS\AppData\Roaming\Nuix\Scripts\NuixProvided\CustodianPstExport.nuixscript\custodian_pst.rb:34)
	at RUBY.<main>(C:\Users\Administrator.GIFS\AppData\Roaming\Nuix\Scripts\NuixProvided\CustodianPstExport.nuixscript\custodian_pst.rb:235)

```

--- a/README.MD
+++ b/README.MD
@@ -1,7 +1,7 @@
 Custodian Export
 ==============
 
-![Last tested in Nuix 7.8](https://img.shields.io/badge/Nuix-7.8-green.svg)
+![Last tested in Nuix 9.0](https://img.shields.io/badge/Nuix-9.0-green.svg)
 
 View the GitHub project [here](https://github.com/Nuix/Custodian-Export) or download the latest release [here](https://github.com/Nuix/Custodian-Export/releases).
 

--- a/Ruby/CustodianPstExport.nuixscript/custodian_pst.rb
+++ b/Ruby/CustodianPstExport.nuixscript/custodian_pst.rb
@@ -1,7 +1,7 @@
 # Menu Title: Custodian PST Export
 # Needs Case: true
 # Needs Selected Items: true
-# @version 2.1.0
+# @version 2.1.1
 
 naming = 'item_name'
 # naming = 'item_name_with_path'
@@ -41,14 +41,13 @@ class CustodianExport < NxProgress
 
   # Creates top-level-MD5-digests.txt by combining per-custodian files.
   def append_digests
-    name = 'top-level-MD5-digests.txt'
-    @@dialog.setSubStatusAndLogIt("Generating #{name}")
-    files = Dir.glob(report_path(File.join('*', name)))
+    @@dialog.setSubStatusAndLogIt('Generating top-level-MD5-digests.txt')
+    files = Dir.glob(report_path(File.join('*', 'top-level-MD5-digests.txt')))
     @@dialog.setSubProgress(0, files.size)
     files.each_with_index do |file, i|
       @@dialog.setSubProgress(i)
       @@dialog.logMessage("Adding #{file}")
-      o = File.new(export_path(name), 'a')
+      o = File.new(File.join(@export_dir, 'top-level-MD5-digests.txt'), 'a')
       o.write(File.read(file))
       o.close
     end

--- a/Ruby/summary_reporter.rb
+++ b/Ruby/summary_reporter.rb
@@ -84,11 +84,16 @@ class SummaryReporter < NxProgress
   # @param file_path [String] path to a summary-report.xml file
   def summarize_file(file_path)
     @@dialog.logMessage("Reading #{file_path}")
-    r = ReportFile.new(file_path)
-    @total_duration += r.duration
-    @configuration = r.configuration if @configuration.nil?
-    @exports << r.details
-    r.statistics.each { |t, v| @stats[t].merge!(v) { |_k, v1, v2| v1 + v2 } }
+    begin
+      r = ReportFile.new(file_path)
+    rescue
+      @@dialog.logMessage("Error loading #{file_path}. Is it XML?")
+    else
+      @total_duration += r.duration
+      @configuration = r.configuration if @configuration.nil?
+      @exports << r.details
+      r.statistics.each { |t, v| @stats[t].merge!(v) { |_k, v1, v2| v1 + v2 } }
+    end
   end
 
   # Generates summary report XML document.

--- a/Ruby/summary_reporter.rb
+++ b/Ruby/summary_reporter.rb
@@ -1,5 +1,5 @@
 # Class for handling summary-report.xml files
-# @version 1.0.0
+# @version 1.0.1
 class SummaryReporter < NxProgress
   require 'rexml/document'
 
@@ -87,7 +87,7 @@ class SummaryReporter < NxProgress
     begin
       r = ReportFile.new(file_path)
     rescue
-      @@dialog.logMessage("Error loading #{file_path}. Is it XML?")
+      @@dialog.logMessage("ERROR loading #{file_path}. Is it XML?")
     else
       @total_duration += r.duration
       @configuration = r.configuration if @configuration.nil?
@@ -132,7 +132,12 @@ class SummaryReporter < NxProgress
   def xml_export
     e = REXML::Element.new 'Export'
     e.add_attributes(export_attributes)
-    export_elements.each { |xml| e << xml }
+    begin
+      export_elements.each { |xml| e << xml }
+    rescue
+      e.add_element('ERROR')
+      @@dialog.logMessage('ERROR generating summary-report.xml')
+    end
     e
   end
 

--- a/Ruby/summary_reporter.rb
+++ b/Ruby/summary_reporter.rb
@@ -134,8 +134,8 @@ class SummaryReporter < NxProgress
     e.add_attributes(export_attributes)
     begin
       export_elements.each { |xml| e << xml }
-    rescue
-      e.add_element('ERROR')
+    rescue => error
+      e.add_element("ERROR: #{error.message}")
       @@dialog.logMessage('ERROR generating summary-report.xml')
     end
     e

